src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/memcache_servers.html.twig

```diff
@@ -36,31 +36,7 @@
 </div>
 
 <div id="server-form" class="collapse">
-    <div class="form-group row">
-        <label class="form-control-label">{{ 'IP Address'|trans }}</label>
-        <div class="col-sm">
-          {{ form_errors(form.memcache_ip) }}
-          {{ form_widget(form.memcache_ip) }}
-        </div>
-    </div>
-
-    <div class="form-group row">
-        <label class="form-control-label">{{ 'Port'|trans }}</label>
-        <div class="col-sm">
-          {{ form_errors(form.memcache_port) }}
-          {{ form_widget(form.memcache_port) }}
-        </div>
-    </div>
-
-    <div class="form-group row">
-        <label class="form-control-label">{{ 'Weight'|trans }}</label>
-        <div class="col-sm">
-          {{ form_errors(form.memcache_weight) }}
-          {{ form_widget(form.memcache_weight) }}
-        </div>
-    </div>
-
-    {{ form_rest(form) }}
+    {{ form_widget(form) }}
```

Comment:
Is it a good practice to use form_widget on the whole form ? I know it works but I never understood why 😅.

IMO it feels more logical to have:
```
form_start(form)

// specific rendering of the widgets you want
form_widget(field1)
form_widget(field2)
form_widget(field3)

// "misc" rendering of whatever is left
form_rest(form)
```

It's even in the name of the function: form_widget aims to render a single widget, form_rest aims to render whatever has not been rendered left (= the rest) 😅?

Reply:
`form_widget` is used to render both the form and its widgets (see [form_widget documentation](https://symfony.com/doc/3.4/reference/forms/twig_reference.html#form-widget-view-variables)).

Moreover, check out what `form()` does: [source code](https://github.com/symfony/twig-bridge/blob/master/Resources/views/Form/form_div_layout.html.twig#L359-L363)

`form_rest` is clearly meant to be used when coupled with several `form_row` in case you forget to render a field.

Reply:
OK, you convinced me 👍